src/browser.ts

```diff
@@ -0,0 +1,214 @@
+ */
+
+import * as protobuf from 'protobufjs';
+import * as gax from './gax';
+import {Status} from './status';
+import {OutgoingHttpHeaders} from 'http';
+import {GoogleAuth} from 'google-auth-library';
+import {OperationsClientBuilder} from './operationsClientBrowser';
+import {GrpcClientOptions, ClientStubOptions} from './grpc';
+import {GaxCall, GRPCCall} from './apitypes';
+import {Descriptor} from './descriptor';
+import {createApiCall as _createApiCall} from './createApiCall';
+
+export {PathTemplate} from './pathTemplate';
+export {CallSettings, constructSettings, RetryOptions} from './gax';
+
+export {
+  BundleDescriptor,
+  LongrunningDescriptor,
+  PageDescriptor,
+  StreamDescriptor,
+} from './descriptor';
+
+export {StreamType} from './streamingCalls/streaming';
+
+export const ALL_SCOPES: string[] = [];
+lro.ALL_SCOPES = ALL_SCOPES;
+
+interface CancelHandler {
+  canceller: AbortController;
+  cancelRequested: boolean;
+}
+
+export class GrpcClient {
+  auth: GoogleAuth;
+  promise?: PromiseConstructor;
+
+  constructor(options: GrpcClientOptions = {}) {
+    if (!options.auth) {
+      throw new Error(
+        'You need to pass auth instance to gRPC-fallback client. Use OAuth2Client from google-auth-library.'
+      );
+    }
+    this.auth = options.auth;
+    this.promise = 'promise' in options ? options.promise! : Promise;
+  }
+
+  loadProto(jsonObject) {
+    const rootObject = protobuf.Root.fromJSON(jsonObject);
+    return rootObject;
+  }
+
+  private getServiceMethods(service: protobuf.Service) {
+    const methods = Object.keys(service.methods);
+
+    // Creating an array of all methods within the stub with correct case (i.e, first letter is lower-case)
+    const methodsCorrectCase = methods.map(method => {
+      return method[0].toLowerCase() + method.substring(1);
+    });
+
+    return methodsCorrectCase;
+  }
+
+  constructSettings(
+    serviceName: string,
+    clientConfig: gax.ClientConfig,
+    configOverrides: gax.ClientConfig,
+    headers: OutgoingHttpHeaders
+  ) {
+    return gax.constructSettings(
+      serviceName,
+      clientConfig,
+      configOverrides,
+      Status,
+      this.promise
+    );
+  }
+
+  async createStub(service: protobuf.Service, opts: ClientStubOptions) {
+    const authHeader = await this.auth.getRequestHeaders();
+    function serviceClientImpl(method, requestData, callback) {
+      let cancelController, cancelSignal;
+      if (typeof AbortController !== 'undefined') {
+        cancelController = new AbortController();
+        cancelSignal = cancelController.signal;
+      }
+      const cancelHandler: CancelHandler = {
+        canceller: cancelController,
+        cancelRequested: false,
+      };
+      const headers = Object.assign({}, authHeader);
+      headers['Content-Type'] = 'application/x-protobuf';
+      headers['User-Agent'] = 'testapp/1.0';
+
+      const grpcFallbackProtocol = opts.protocol || 'https';
+      let servicePath = opts.servicePath;
```

Comment:
honestly I think `servicePath` is a poor name for the `host`. Its confusing, but I realize that the original option here (`opts.servicePath`) is named as such.

Reply:
It's poor but it's used everywhere in the implementation. Good news is that users can also set `apiEndpoint` (and it will be automatically converted to `servicePath` in the autogen'd client library code, e.g. here https://github.com/googleapis/nodejs-dlp/blob/f9315d1eddb3e98248784cfac27fd94f0231be46/src/v2/dlp_service_client.js#L70)

Reply:
OK. I will resolve as a no-op because its standard in the repo.